p2p/service.py

```diff
@@ -148,6 +149,7 @@ def run_daemon_task(self, awaitable: Awaitable[Any]) -> None:
         Like :meth:`run_task` but if the task ends without cancelling, then this
         this service will terminate as well.
         """
+        @functools.wraps(awaitable)  # type: ignore
```

Comment:
I think this is sufficient since it appears that the `__name__` is preserved on the couroutine object.  It however doesn't preserve things like `__doc__` but I think that is acceptable.